llvm/test/Feature/Repo/repo_pruning_llvm.global_ctors.ll

```diff
@@ -0,0 +1,55 @@
+; This testcase tests the program repository pruning pass for the llvm.global_ctors.
```

Comment:
This says that it tests the "program repository pruning pass": does it? (The change in this CR is to ElimAvailExtern.cpp which isn't part of the pruning code.) I'm slightly nervous that this test isn't looking at the modified optimization pass but at the system as a whole. Can we test the behaviour of the modified pass in isolation?

This suggests that the fix is specific to the llvm.global_ctors variable whereas the change to ElimAvailExtern.cpp above doesn't appear to contain anything specific to any particular variable.

Can the test be simplified to remove unnecessary code and to eliminate use of the "special" variables like llvm.global_ctors?

Reply:
> 
> Can we test the behaviour of the modified pass in isolation?
>
Good point. A new test named `repo_eliminate_available_externally_globals.ll` has been added.

> 
> Can the test be simplified to remove unnecessary code and to eliminate use of the "special" variables like llvm.global_ctors?
>
Yes, the test has been renamed and updated following your comments.